src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -702,6 +702,12 @@ public virtual DeleteLinkUriOption DeleteLinkUriOption
             set { this.deleteLinkUriOption = value; }
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether collections of untyped values should be supported in response payloads; if they are not supported, an exception will
+        /// be thrown when they are encountered
+        /// </summary>
+        public virtual bool EnableUntypedCollectionResponses { get; set; } = true;

```

Comment:
I defaulted to `false` in the `ODataMessageReaderSettings` since a caller might be relying on the parsed type being `ODataIUntypedValue`, but for the `DataServiceContext` I think it's unlikely that a caller is relying on an exception for a perfectly valid model + response combination. I'm open to other interpretations, though. What do you think?

Reply:
Talking with Mike, I'm planning to remove this behavior flag, but keep the one for `ODataMessageReaderSettings`.

Reply:
Will the customers using `DataServiceContext` be able to configure the `ODataMessageWriterSettings`?

Reply:
We will set ODataMessageReaderSettings.EnableUntypedCollections to true and they won't have the option to set it to false (I just updated the code to actually reflect this). However, looking this morning, they still can have the option to disable it if they truly need to by using the DataServiceContext.Configurations.ResponsePipeline. I updated the one test to use this mechanism to exemplify it and confirm it works.